### ISSUE TITLE
How to group by one tree?

### ISSUE THREAD

**Issue by ohohco** [ceolter/angular-grid#163]
My data is a tree struct, all only a parent root node,
for  example:
```
[
  {id:1,pid:0,name:'root'},
  {id:2,pid:1,name:'oneLevel1'},
  {id:3,pid:1,name:'oneLevel2'},
  {id:4,pid:2,name:'twoLevel1'},
 {id:5,pid:2,name:'twoLevel2'},
 {id:6,pid:5,name:'threeLevel1'},
...
]
```
How to render more levels like this:
<pre>
root
|-----oneLevel1
       |-------twoLevel1
|-----oneLevel2
       |-------twoLevel2
                |-------threeLevel1
</pre>

**Comment by ohohco** [comment]
I see fileBrowser.js can implements ,but need to process the data.  Will to be built-in angular-grid？

**Comment by ceolter** [comment]
you either create the data structure yourself (as in the file browser example) or you get the grid to group for you, where you group by particular attributes in your data.

for the second, your data needs to be in a style similar to:
[
 {id: 1, dim1: 'root', dim2: 'a', dim3: 'u', value: 'bla'},
 {id: 1, dim1: 'root', dim2: 'a', dim3: 'v', value: 'bla'},
 {id: 1, dim1: 'root', dim2: 'a', dim3: 'w', value: 'bla'},
 {id: 1, dim1: 'root', dim2: 'b', dim3: 'x', value: 'bla'},
 {id: 1, dim1: 'root', dim2: 'b', dim3: 'y', value: 'bla'},
 {id: 1, dim1: 'root', dim2: 'b', dim3: 'z', value: 'bla'},
]

then get the grid to group by dim1, dim2 and dim3

**Comment by ohohco** [comment]
I structure the data like the brower example from the ajax, if I set rowsAlreadyGrouped=true, will error.else there is no view in grid

**Comment by ceolter** [comment]
sounds like there is something wrong in your code. and i don't have enough  info to help you :(

can you post this on the bulletin board? the issues for github should be for errors in the grid!!

maybe post the stacktrace and error message? or can you recreate the problem in a fiddle? that way i can debug it!

**Comment by ohohco** [comment]
Same as fileBrowser.js ,just use data by ajax.
```
$scope.gridOptions = {
        rowData: null, //change
        ....
    };
//by ajax
$http.get("/xxx")
.then(function(resp) {
	$scope.gridOptions.rowData = rowData;
	$scope.gridOptions.api.onNewRows();
});
``` 
In browser console:
```
TypeError: Cannot read property 'length' of null
    at r.recursivelyCheckUserProvidedNodes (angularGrid.min.js:2)
    at r.setAllRows (angularGrid.min.js:2)
    at new r (angularGrid.min.js:1)
    at new r (angularGrid.min.js:1)
    at e (angular.js:4045)
    at Object.instantiate (angular.js:4056)
    at $get (angular.js:7391)
    at angular.js:6761
    at q (angular.js:332)
    at A (angular.js:6748)
```

**Comment by ceolter** [comment]
Your data is not structured right. Message implies it's expecting an array
(with length)  but doesn't find.

**Comment by ohohco** [comment]
This  `$scope.gridOptions.rowData = rowData`, `rowData` is the define in the fileBrowser.js

**Comment by ohohco** [comment]
I get the reason, in gridOption should set `rowData:[]` init...

**Comment by ceolter** [comment]
oh dear :( i'm sorry!!! i didn't catch that one. although you have it fixed on your end, i've fixed it up this end also, will be in 1.7.0, so won't be a problem for the next person. thanks :)
